src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -16,10 +16,25 @@
 
 namespace Microsoft.Data.SqlClient
 {
+    internal readonly ref struct SessionHandle
+    {
+        public readonly SNIHandle NativeHandle;
+
+        public SessionHandle(SNIHandle nativeHandle) => NativeHandle = nativeHandle;
+        
+        public bool IsNull => NativeHandle is null;
+    }
+
     internal partial class TdsParserStateObject
     {
+        private static bool UseManagedSNI => false;
+
         private SNIHandle _sessionHandle = null;              // the SNI handle we're to work on
 
+        private SessionHandle SessionHandle => new SessionHandle(_sessionHandle);
+
+        private bool TransparentNetworkIPResolution => _parser.Connection.ConnectionOptions.TransparentNetworkIPResolution;
+
         internal bool _pendingData = false;
```

Comment:
Could you make private this field and use HasPendingData property where it's not used

Reply:
These fields are used across many files, that's why I didn't touch them. But I now see they will help with merging TdsParser, so I will add them.

Reply:
Yes that was my point to change them and it's a quick harmless refactoring. Thanks

Reply:
Just added a commit implementing the requested change.